From efb693cbc71c977c33e057e5a799471504461ba9 Mon Sep 17 00:00:00 2001
From: matsi <matsi@localhost>
Date: Wed, 9 Feb 2011 23:59:23 +0300
Subject: [PATCH 4/4] Modified FPGA map

Now we use CS4 and CS5 to access FPGA:
- CS4: video
- CS5: control
---
 arch/arm/mach-at91/board-lanc.c |   30 ++++++++++++++++++------------
 1 files changed, 18 insertions(+), 12 deletions(-)

diff --git a/arch/arm/mach-at91/board-lanc.c b/arch/arm/mach-at91/board-lanc.c
index 0ea120c..6897200 100644
--- a/arch/arm/mach-at91/board-lanc.c
+++ b/arch/arm/mach-at91/board-lanc.c
@@ -169,29 +169,35 @@ static void __init lanc_add_device_fpga(void)
 	at91_sys_write(AT91_MATRIX_EBICSA, ebi0_csa);
 #endif
 
-	/* Configure SMC on CS1 */
+	/* Configure SMC on CS4 */
 	csa = at91_sys_read(AT91_MATRIX_EBICSA);
-	csa |= AT91_MATRIX_EBI_CS1A_SMC;
+	csa |= AT91_MATRIX_EBI_CS4A_SMC;
 	at91_sys_write(AT91_MATRIX_EBICSA, csa);
 
-	at91_sys_write(AT91_SMC_SETUP(1), AT91_SMC_NWESETUP_(0) | AT91_SMC_NCS_WRSETUP_(1) |
+	at91_set_A_periph(AT91_PIN_PC10, 0);
+
+	at91_sys_write(AT91_SMC_SETUP(4), AT91_SMC_NWESETUP_(0) | AT91_SMC_NCS_WRSETUP_(1) |
 					   AT91_SMC_NRDSETUP_(7) | AT91_SMC_NCS_RDSETUP_(7));
-	at91_sys_write(AT91_SMC_PULSE(1), AT91_SMC_NWEPULSE_(3) | AT91_SMC_NCS_WRPULSE_(2) |
+	at91_sys_write(AT91_SMC_PULSE(4), AT91_SMC_NWEPULSE_(3) | AT91_SMC_NCS_WRPULSE_(2) |
 					   AT91_SMC_NRDPULSE_(3) | AT91_SMC_NCS_RDPULSE_(43));
-	at91_sys_write(AT91_SMC_CYCLE(1), AT91_SMC_NWECYCLE_(8) | AT91_SMC_NRDCYCLE_(62));
+	at91_sys_write(AT91_SMC_CYCLE(4), AT91_SMC_NWECYCLE_(8) | AT91_SMC_NRDCYCLE_(62));
+
+	at91_sys_write(AT91_SMC_MODE(4), AT91_SMC_PMEN | AT91_SMC_TDFMODE | AT91_SMC_DBW_16 | AT91_SMC_BAT_WRITE );
 
-	at91_sys_write(AT91_SMC_MODE(1), AT91_SMC_PMEN | AT91_SMC_TDFMODE | AT91_SMC_DBW_16 | AT91_SMC_BAT_WRITE );
+	/* Configure SMC on CS5 */
+	csa = at91_sys_read(AT91_MATRIX_EBICSA);
+	csa |= AT91_MATRIX_EBI_CS5A_SMC;
+	at91_sys_write(AT91_MATRIX_EBICSA, csa);
 
-	/* Configure SMC on CS2 */
-	at91_set_A_periph(AT91_PIN_PC13, 0);
+	at91_set_A_periph(AT91_PIN_PC11, 0);
 
-	at91_sys_write(AT91_SMC_SETUP(2), AT91_SMC_NWESETUP_(0) | AT91_SMC_NCS_WRSETUP_(1) |
+	at91_sys_write(AT91_SMC_SETUP(5), AT91_SMC_NWESETUP_(0) | AT91_SMC_NCS_WRSETUP_(1) |
 					   AT91_SMC_NRDSETUP_(7) | AT91_SMC_NCS_RDSETUP_(7));
-	at91_sys_write(AT91_SMC_PULSE(2), AT91_SMC_NWEPULSE_(3) | AT91_SMC_NCS_WRPULSE_(2) |
+	at91_sys_write(AT91_SMC_PULSE(5), AT91_SMC_NWEPULSE_(3) | AT91_SMC_NCS_WRPULSE_(2) |
 					   AT91_SMC_NRDPULSE_(3) | AT91_SMC_NCS_RDPULSE_(43));
-	at91_sys_write(AT91_SMC_CYCLE(2), AT91_SMC_NWECYCLE_(8) | AT91_SMC_NRDCYCLE_(62));
+	at91_sys_write(AT91_SMC_CYCLE(5), AT91_SMC_NWECYCLE_(8) | AT91_SMC_NRDCYCLE_(62));
 
-	at91_sys_write(AT91_SMC_MODE(2), AT91_SMC_PMEN | AT91_SMC_TDFMODE | AT91_SMC_DBW_16 | AT91_SMC_BAT_WRITE );
+	at91_sys_write(AT91_SMC_MODE(5), AT91_SMC_PMEN | AT91_SMC_TDFMODE | AT91_SMC_DBW_16 | AT91_SMC_BAT_WRITE );
 }
 
 
-- 
1.7.3.5

